bbid.py

```diff
@@ -36,12 +39,15 @@ def download(url,output_dir,retry=False):
 		md5 = hashlib.md5()
 		md5.update(image)
 		md5_key = md5.hexdigest()
+		img_hash_lock.acquire()
 		if md5_key in image_md5s:
```

Comment:
1) Without a lock,
Two or more threads with the same image (or md5 hash) can pass this conditional statement.
2) As a result, two duplicated downloads will be generated